pkg/server/backend_manager.go

```diff
@@ -293,8 +293,8 @@ func ignoreNotFound(err error) error {
 
 // GetRandomBackend returns a random backend connection from all connected agents.
 func (s *DefaultBackendStorage) GetRandomBackend() (Backend, error) {
-	s.mu.RLock()
```

Comment:
This function is only reading though, can you expand on why we need write lock too?

Reply:
@andrewsykim Please see the description in the PR. Without that the test fails due to race conditions.

But here is what I think:
```
  math/rand.(*rngSource).Uint64()
      /snap/go/9028/src/math/rand/rng.go:244 +0xbc
  math/rand.(*rngSource).Int63()
      /snap/go/9028/src/math/rand/rng.go:234 +0x1f1
  math/rand.(*Rand).Int63()
      /snap/go/9028/src/math/rand/rand.go:84 +0xa8
  math/rand.(*Rand).Int31()
      /snap/go/9028/src/math/rand/rand.go:98 +0xb3
  math/rand.(*Rand).Int31n()
      /snap/go/9028/src/math/rand/rand.go:130 +0xae
  math/rand.(*Rand).Intn()
```
If you check the Uint64() function ( /snap/go/9028/src/math/rand/rng.go:244), there is a decrement operation:
```
	rng.feed--
```

That's why the race detector alerts.

Reply:
> Please see the description in the PR. Without that the test fails due to race conditions.

Yeah, sorry I wasn't doubting that we need this, the test result was really useful. Just trying to understand why the read lock is not sufficient if this function is only reading values.

Reply:
Sure, no problem. Thank you for cheching the PR!